src/jit/codegenxarch.cpp

```diff
@@ -982,6 +982,13 @@ void CodeGen::genCodeForMul(GenTreeOp* treeNode)
             unsigned int scale = (unsigned int)(imm - 1);
             getEmitter()->emitIns_R_ARX(INS_lea, size, targetReg, rmOp->gtRegNum, rmOp->gtRegNum, scale, 0);
         }
+        else if (!requiresOverflowCheck && rmOp->isUsedFromReg() && isPow2(imm))
+        {
+            // Use shift for constant multiply when legal
+            unsigned int shiftAmount = genLog2((uint64_t)imm);
```

Comment:
> Why `uint64_t` and not `size_t`? 

I thought it best to exactly match one of the [overloads](https://github.com/dotnet/coreclr/blob/7ca42d2be181b66439220ef98bd0df97b565216b/src/jit/compiler.h#L128-L129) of `genLog2`.

> a static_cast would be preferable to a C style cast

Indeed; updated.